internal/http/types.go

```diff
@@ -131,6 +132,27 @@ type AutoReleasePolicy struct {
 	Environment string `json:"environment,omitempty"`
 }
 
+type BranchRestrictorPolicy struct {
+	ID            string `json:"id,omitempty"`
+	Environment   string `json:"environment,omitempty"`
+	BranchMatcher string `json:"branchMatcher,omitempty"`
+}
+
+type ApplyBranchRestrictorPolicyRequest struct {
+	Service        string `json:"service,omitempty"`
+	Environment    string `json:"environment,omitempty"`
+	BranchMatcher  string `json:"branchMatcher,omitempty"`
+	CommitterName  string `json:"committerName,omitempty"`
+	CommitterEmail string `json:"committerEmail,omitempty"`
+}
```

Comment:
Consider a `PolicyMetadata` to share it between policies?

These seem shared both conceptually and functionality-wise:
* Service
* Environment
* CommitterName
* CommitterEmail

Reply:
I think I'll wait with this and see if it gets more clear later on. I can see the overlap of the two right now 👍

Reply:
Sounds good with too. Just wanted to mention it 😊